### PR TITLE
Vishwajit/patch to downgrade iap sdk

### PR DESCRIPTION
**Pull request by vishwajitshep0815**
## Summary

Downgraded IAP SDK version 1.4.0

### CODE REVIEW

File: RNSquareInAppPayments.podspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 Pod::Spec.new do |s|
   s.name         = "RNSquareInAppPayments"
-  s.version      = "1.5.0"
+  s.version      = "1.4.0"
```
Comment (hukid) [inline RNSquareInAppPayments.podspec:4]: We don't change this version, instead, it should be 1.5.1

File: RNSquareInAppPayments.podspec [hunk 2]
```diff
@@ -17,7 +17,7 @@ Pod::Spec.new do |s|
   s.resource_bundle = { "RNSquareInAppPayments-Resources" => ["ios/RNSquareInAppPayments-Resources/*.lproj/*.strings"] }
 
   s.dependency "React"
-  s.dependency "SquareInAppPaymentsSDK", '1.5.1'
-  s.dependency 'SquareBuyerVerificationSDK', '1.5.1'
+  s.dependency "SquareInAppPaymentsSDK", '1.4.0'
+  s.dependency 'SquareBuyerVerificationSDK', '1.3.0'
```
Comment (hukid) [inline RNSquareInAppPayments.podspec:21]: Please use `1.4.0`, it was missing but it should work now.

**Review by hukid — CHANGES_REQUESTED** [review]

File: RNSquareInAppPayments.podspec (hunk 1 above)
Comment (vishwajitshep0815) [reply RNSquareInAppPayments.podspec]: It's done.

File: RNSquareInAppPayments.podspec (hunk 2 above)
Comment (vishwajitshep0815) [reply RNSquareInAppPayments.podspec:21]: I tried changing SquareBuyerVerificationSDK (= 1.3.0) to SquareBuyerVerificationSDK (= 1.4.0) but the pod install fails with the following error.
[!] CocoaPods could not find compatible versions for pod "SquareBuyerVerificationSDK":
  In snapshot (Podfile.lock):
    SquareBuyerVerificationSDK (= 1.3.0)

  In Podfile:
    RNSquareInAppPayments (from `../node_modules/react-native-square-in-app-payments`) was resolved to 1.5.1, which depends on
      SquareBuyerVerificationSDK (= 1.4.0)

Specs satisfying the `SquareBuyerVerificationSDK (= 1.3.0), SquareBuyerVerificationSDK (= 1.4.0)` dependency were found, but they required a higher minimum deployment target.

I tried changing the deployment target to various version and the latest one but that wasn't successful either.
I also checked pod install with 1.5.1 which was successful but for that need to update both the SDK versions to 1.5.1 on iOS.
    - SquareBuyerVerificationSDK (= 1.5.1)
    - SquareInAppPaymentsSDK (= 1.5.1)

**Comment by w0o** [comment]
Interestingly this might help with this issue:
https://github.com/square/in-app-payments-react-native-plugin/issues/116

**Comment by w0o** [comment]
Interestingly this might help with https://github.com/square/in-app-payments-react-native-plugin/issues/116

**Review by hukid — APPROVED** [review]

File: android/build.gradle [hunk 3]
```diff
@@ -12,7 +12,7 @@ buildscript {
 
 apply plugin: 'com.android.library'
 
-def MIN_IAP_SDK_VERSION = '1.5.1'
```
Comment (Ruby-star-cloud) [inline android/build.gradle:15]: - # - [ ] ~~_****_~~

**Review by Ruby-star-cloud — APPROVED** [review]

**Pull request merged**